### PR TITLE
Add subcategories for Bug feedback type

### PR DESCRIPTION
**Pull request by kahrendsen-li**
Adds the ability to optionally set a list of subcategories for a feedback type, and uses this capability for the Bug feedback type to allow specifying if the bug was a crash or non-fatal.

### CODE REVIEW

File: shaky/src/main/res/values/shaky_strings.xml [hunk 1]
```diff
@@ -63,4 +63,11 @@
 
     <string name="shaky_setting">Settings</string>
     <string name="shaky_general_ok_string">OK</string>
+
+    <string name="shaky_select_subcategory">Select a subcategory</string>
+    <string-array name="shaky_bug_subcategories">
+        <item>Crash</item>
+        <item>Non-Fatal</item>
```
Comment (csv8674xn) [inline shaky/src/main/res/values/shaky_strings.xml:70]: Not sure if Non-Fatal is common enough to use it externally.
Isn't non-fatal just a regular bug?

File: shaky/src/main/java/com/linkedin/android/shaky/FormFragment.java [hunk 2]
```diff
@@ -133,20 +178,80 @@ public boolean onMenuItemClick(MenuItem item) {
...
-            alertDialog.show();
+            showAlertDialog(getString(R.string.shaky_empty_feedback_message));
             return false;
         }
+        return true;
+    }
 
+    private boolean validateSubcategory(int selectedItemPosition) {
+        if (selectedItemPosition == 0) {
+            showAlertDialog(getString(R.string.shaky_subcategory_not_selected_message));
```
Comment (csv8674xn) [inline shaky/src/main/java/com/linkedin/android/shaky/FormFragment.java:191]: The only difference of validate and validateMessage is checking the empty string case. we can just reuse the original validate method without creating a new helper function here. Showing an empty dialog to user is also not what we want

File: shaky/src/main/java/com/linkedin/android/shaky/FormFragment.java [hunk 3]
```diff
@@ -44,23 +49,44 @@
...
+     *                 order as the subtype labels array from the "subtypeLabels" param
+     * @return A new {@link FormFragment}
+     */
     public static FormFragment newInstance(@NonNull String title,
                                            @NonNull String hint,
                                            @Nullable Uri screenshotUri,
-                                           @Nullable @MenuRes int menu) {
+                                           @Nullable @MenuRes int menu,
+                                           @Nullable @ArrayRes int subtypeLabels,
+                                           @Nullable String[] subtypes) {
```
Comment (csv8674xn) [inline shaky/src/main/java/com/linkedin/android/shaky/FormFragment.java:82]: I will not try to modify this public function, which introduces a breaking change on all downstream callers. Another newInstance method and delegate the call to this one should help

Comment (kahrendsen-li) [reply shaky/src/main/java/com/linkedin/android/shaky/FormFragment.java]: Done

File: shaky/src/main/java/com/linkedin/android/shaky/FormFragment.java (hunk 2 above)
Comment (kahrendsen-li) [reply shaky/src/main/java/com/linkedin/android/shaky/FormFragment.java:208]: Discussed offline, agreed to keep the two functions separate

File: shaky/src/main/res/values/shaky_strings.xml (hunk 1 above)
Comment (kahrendsen-li) [reply shaky/src/main/res/values/shaky_strings.xml]: Changed to a more friendly string

File: shaky/src/main/java/com/linkedin/android/shaky/FormFragment.java [hunk 4]
```diff
@@ -133,20 +195,80 @@ public boolean onMenuItemClick(MenuItem item) {
...
+            add(UNSELECTED);
+            addAll(subtypes);
+        }
+
+        @Override
+        public View getView(int position, @Nullable View convertView,
+            @NonNull ViewGroup parent) {
+            TextView textView = makeTextView(adapterLabels[position], convertView, parent,
+                R.layout.shaky_spinner_item);
+            return textView;
```
[the remark this answers is not in the document]
Comment (kahrendsen-li) [reply shaky/src/main/java/com/linkedin/android/shaky/FormFragment.java]: Done

File: shaky/src/main/java/com/linkedin/android/shaky/FormFragment.java [hunk 5]
```diff
@@ -133,20 +195,80 @@ public boolean onMenuItemClick(MenuItem item) {
...
+            new DialogInterface.OnClickListener() {
+                public void onClick(DialogInterface dialog, int which) {
+                    dialog.dismiss();
+                }
+            });
+        alertDialog.show();
+    }
+
+    private class FeedbackSubtypeAdapter extends ArrayAdapter<String> {
+        CharSequence[] adapterLabels;
```
[the remark this answers is not in the document]
Comment (kahrendsen-li) [reply shaky/src/main/java/com/linkedin/android/shaky/FormFragment.java]: Done

**Pull request merged**